src/town_gui.cpp

```diff
@@ -689,8 +693,8 @@ struct TownDirectoryWindow : public Window {
 	/** Sort by town name */
 	static bool TownNameSorter(const Town * const &a, const Town * const &b)
 	{
-		static char buf_cache[64];
-		char buf[64];
+		static char buf_cache[MAX_LENGTH_TOWN_NAME_CHARS];
+		char buf[MAX_LENGTH_TOWN_NAME_CHARS];
```

Comment:
i feel like these should be `MAX_LENGTH_TOWN_NAME_CHARS * MAX_CHAR_LENGTH`, same as the QueryString - a townname could conceivably be constructed out of 64 4-byte characters, and of course "char" is just a byte

Ideally the 64 -> constant changes should be in a separate commit (there may be other instances of this being incorrect as well)

Reply:
I changed them and seperated commit as you said, thanks a lot.